### PR TITLE
[AI] Randomise greet message on first turn

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
- move greeting phrase to stringtables
- add different greetings to different AI

Need help to add more phrases.

PS AI_? number and order of greetings does not match.

### CODE REVIEW

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -14633,7 +14633,12 @@ MONSTER_NAMES
 [[SHIP_NAME_UNSORTED_MYTHOLOGY]]
 [[SHIP_NAME_DEVS]]'''
 
-
+AI_FIRST_TURN_GREETING
+'''Ave, Human, morituri te salutant!
+Get out of our universe!
+Prepare to die.
+8008135
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: no

Comment (Vezzra) [inline default/stringtables/en.txt]: Are you objecting specifically to the "8008135" line, or all those additional greetings in general?

**Comment by Vezzra** [comment]
Aside from the discussion about the specific lines, the basic idea is good.

File: default/stringtables/en.txt (hunk 1 above)
Comment (geoffthemedio) [inline default/stringtables/en.txt]: that line

**Comment by geoffthemedio** [comment]
Better would be aggression-dependent lists

**Comment by Vezzra** [comment]
Yep, that sounds even better :smiley:

File: default/stringtables/en.txt (hunk 1 above)
Comment (adrianbroher) [inline default/stringtables/en.txt]: Andrey pls…

[Nice try pal.](https://wiki.teamfortress.com/w/images/a/ae/Scout_beingshotinvincible05.wav)

**Comment by Cjkjvfnby** [comment]
Will remove that line from history.

I lack of ideas. Need at least same amount of phrases as default number of AIs.

Making aggression dependent means that we need more phrases.

**Comment by MatGB** [comment]
Get out of our _universe_ needs to be _galaxy_, and I so want to insert "the hell" after get, but probably shouldn't, we have enough Bab5 references already.

OK, we want 6ish phrases for each aggression setting? That's doable. It definitely needs to be in the stringtable for future use, it was just a quick hack to confirm the AI was working initially.

**Comment by Vezzra** [comment]
Some suggestions:

```
To do battle with you will be an honor.
Alien scum! We shall cleanse the galaxy of your filthy presence!
Worthless insects, you dare to challenge our mighty empire?!
Exterminating you shall be a genuine pleasure!
Welcome, dear enemy, our fearless warriors are waiting for you...
Don't bother to look for mercy to us, for you will find none here.
A meek lamb who wants to dance with lions is always welcome at our table - as the entrée of course...
Ha! You think you can stand up to us? Well, bring it on!
A bunch of foolish aliens stumbling right into our arms - a day can't start better!
Despair and defeat is what you're going to find at our hands.
Greetings strangers! We hope you are worthy doing battle with.
Surrender now, and we might consider letting you live... as our slaves forever.
Enjoy yourself as long as you can, for the last thing you'll see is the cleansing fire of our powerful weapons!
Sending you to your ancestors is our sacred duty!
```

To our native speakers: are those somewhat acceptable? I hope they don't sound too silly... feel free to suggest corrections.

**Comment by Vezzra** [comment]
> we have enough Bab5 references already

What's wrong with Babylon 5 references? :wink:  "Get the hell out of our galaxy!" sounds too cool to pass over...

**Comment by MatGB** [comment]
Nothing, at all, but, y'know, direct quotes might be pushing it legally? Not my field really.

We probably need a few cowardly/isolationist ones. "Please leave us alone?"

File: default/AI/FreeOrionAI.py [hunk 2]
```diff
@@ -209,7 +209,9 @@ def generateOrders():  # pylint: disable=invalid-name
     if turn == 1:
         declare_war_on_all()
         human_player = fo.empirePlayerID(1)
-        fo.sendChatMessage(human_player,  '%s Empire (%s):\n"Ave, Human, morituri te salutant!"' % (empire.name, aggression_name))
+        greets = UserStringList("AI_FIRST_TURN_GREETING")
```
Comment (Vezzra) [inline default/AI/FreeOrionAI.py]: I suggest to add a `random.shuffle(greets)` after that, so the order gets randomized (don't want to get the same greetings on every map :wink:).

**Comment by Vezzra** [comment]
> Nothing, at all, but, y'know, direct quotes might be pushing it legally?

Honestly, the day I've to worry about potential legal issues over a generic phrase like "Get the hell out of our galaxy!" is the day when I finally quit, throw all things digital and electronic out of my window and retire to a remote, lonely island, not to be bothered by mankinds moronic antics ever again...

:wink:

**Comment by Vezzra** [comment]
> We probably need a few cowardly/isolationist ones. "Please leave us alone?"

Hm, "Stay the hell away from us!"? Lets see:

```
Don't bother with us, we have nothing to offer that might be of interest for you.
Stay away from us, and we will stay away from you!
We won't bother you, but if you don't leave us alone, you shall find us well prepared to defend ourselves!
What have we done to deserve your wrath, oh mighty ones?
Please spare your humble servants, and our gratitude shall be forever yours!
In both of our interests, let us leave each other alone.
```

**Comment by Cjkjvfnby** [comment]
Updated:
- add random choose of message
- add choose of phrases dependent on aggression (with beginner as default)
- add strings by Vezzra

PS Feel free to add commits with changes via web interface. I can miss something in this dialog.

**Comment by AStrayWitch** [comment]
Changing "Don't bother to look for mercy to us, for you will find none here" to "Don't bother to look to us for mercy, for you will find none here",or "mercy from us" would be better grammar,and(i think) sound better.

**Comment by Vezzra** [comment]
@isngrim: thanks, correction incorporated.

I've added the rest of my suggestions, moved the entire AI greetings lists to the AI diplomatic messages section of the stringtable, added greetings lists for the rest of the aggression settings and reassigned several of the greetings to have at least a few in each list.

We still need more greetings (especially for the beginner, turtle, cautious and typical lists), but even so it's already far better than what we had before. So I think this can be merged, unless anyone has objections?

**Comment by geoffthemedio** [comment]
I suppose that if this is implemented using a UserStringList and picking one of the already-translated entries from it, then it's not possible to just send the [[STRINGTABLE_KEY]] to the player for client-side translation. Avoiding assuming a particular language within the AI player would be preferable, though.

**Comment by Vezzra** [comment]
Well, for the default use case, single player with couple of AIs, the current approach still works, as the string list should be picked from the correct stringtable. Of course, a multiplayer game with AI players is a different story.

But for that use case we'd need to revise the current simple string list design. The proper way to implement them would probably be to remove them from the stringtables entirely, put them into their own file, and have them contain stringtable keys, not already translated entries.

**Comment by geoffthemedio** [comment]
I suppose you could put a bunch of stringtable keys into the list, and have separate stringtable entries for each of them, and then the AI client would send

```
[[KEY_FROM_LIST]]
```

in chat...

And there's no way to do client-side stringtable substitutions in chat now. Would need a VarText essentially (like a sitrep).

**Comment by Vezzra** [comment]
> I suppose you could put a bunch of stringtable keys into the list, and have separate stringtable entries for each of them

Sure, that will certainly work, but I think the cleaner, more consistent solution would be to remove everything that doesn't require translation, or better, must not be translated out of the stringtables and put into a separate file. We have already a few of those "functional stringlists" (which are accompanied by warning comments to _not_ translate them, so translators don't mess them up), the greetings lists are just one more.

Of course, the implementation for that will be a bit more complicated than just using the userstring functions.

**Comment by geoffthemedio** [comment]
I don't think I understand what you're proposing then.

**Comment by Vezzra** [comment]
Well, maybe I misunderstood your reply, but it sounded like you want to keep the greetings stringlists in the stringtable files, just instead of having the already translated entries in the greetings lists put stringtable keys there and of course add stringtable entries for those keys too.

As a quick stop-gap solution that's ok. My proposal is merely that on the long run we should consider taking all those "functional stringlists" (there are others, e.g. the one which determines the order of sitrep messages) into a separate file. After all, those functional lists must not be translated anyway, so they don't actually belong in the stringtables.

**Comment by Vezzra** [comment]
Hm, on a second thought, regarding translation of chat messages: what if we devise an approach that works the other way round? Instead of sending stringlist keys (and maybe even a list of parameters), do the entire composition of the text in the correct language on the senders side. This will require three things:
- The clients need to be able to query the language settings of the other clients
- The clients need to be able to optionally specify the stringtable when calling `UserString`
- Each client must have the same set of stringtables (which usually should be the case anyway)

(Just a thought.)

**Comment by genestack-solomatin** [comment]
We got most important part: content. 
It work.
Lets merge.

All improvements we can do later.

**Comment by Vezzra** [comment]
> I suppose you could put a bunch of stringtable keys into the list, and have separate stringtable entries for each of them, and then the AI client would send [[KEY_FROM_LIST]] in chat...

Implemented that suggestion: 2902aed12fbef67ee2268e25bfa143bd537dd745

**Pull request merged**